src/states/elder/mod.rs

```diff
@@ -489,33 +493,40 @@ impl Elder {
 
     fn dispatch_routing_message(
         &mut self,
-        routing_msg: RoutingMessage,
+        msg: SignedRoutingMessage,
         outbox: &mut dyn EventBox,
-    ) -> Result<(), RoutingError> {
+    ) -> Result<Transition, RoutingError> {
         use crate::messages::MessageContent::*;
 
-        match routing_msg.content {
+        let (msg, metadata) = msg.into_parts();
+
+        match msg.content {
             UserMessage { .. } => (),
-            _ => trace!("{} Got routing message {:?}.", self, routing_msg),
+            _ => trace!("{} Got routing message {:?}.", self, msg),
```

Comment:
reusing `msg` in `let (msg, metadata) = msg.into_parts();` is bit confusing, I'd prefer have a new name ?